modules/userdirectory-brightspace/src/main/java/org/opencastproject/userdirectory/brightspace/BrightspaceUserProviderInstance.java

```diff
@@ -0,0 +1,274 @@
+          int cacheExpiration) {
+    this.pid = pid;
+    this.client = client;
+    this.organization = organization;
+    logger.info("Creating new BrightspaceUserProviderInstance(pid={}, url={}, cacheSize={}, cacheExpiration={})", pid,
+            client.getURL(), cacheSize, cacheExpiration);
+
+    cache = CacheBuilder.newBuilder().maximumSize(cacheSize).expireAfterWrite(cacheExpiration, TimeUnit.MINUTES)
+            .build(new CacheLoader<String, Object>() {
+              @Override
+              public Object load(String username) {
+                User user = loadUserFromBrightspace(username);
+                return user == null ? nullToken : user;
+              }
+            });
+
+    this.registerMBean(pid);
+  }
+
+  @Override
+  public float getCacheHitRatio() {
+    if (loadUserRequests.get() == 0)
+      return 0;
+    return (float) (loadUserRequests.get() - brightspaceWebServiceRequests.get()) / loadUserRequests.get();
+  }
+
+  private void registerMBean(String pid) {
+    this.loadUserRequests = new AtomicLong();
+    this.brightspaceWebServiceRequests = new AtomicLong();
+
+    try {
+      ObjectName name = BrightspaceUserProviderFactory.getObjectName(pid);
+      MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+
+      try {
+        mbs.unregisterMBean(name);
+      } catch (InstanceNotFoundException ife) {
+        logger.debug("{} was not registered", name);
+      }
+
+      mbs.registerMBean(this, name);
+    } catch (Exception e) {
+      logger.error("Unable to register {} as an mbean: {}", this, e);
+    }
+
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#getName()
+   */
+  @Override
+  public String getName() {
+    return pid;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#getUsers()
+   */
+  @Override
+  public Iterator<User> getUsers() {
+    return Collections.emptyIterator();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#loadUser(java.lang.String)
+   */
+  @Override
+  public User loadUser(String userName) {
+    this.loadUserRequests.incrementAndGet();
+    logger.debug("getting user from cache");
+
+    try {
+      Object user = this.cache.getUnchecked(userName);
+      if (user != this.nullToken) {
+        logger.debug("Returning user {} from cache", userName);
+        return (User) user;
+      } else {
+        logger.debug("Returning null user from cache");
+        return null;
+      }
+    } catch (ExecutionError | UncheckedExecutionException ee) {
+      logger.warn("Exception while loading user {}", userName, ee);
+      return null;
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#countUsers()
+   */
+  @Override
+  public long countUsers() {
+    return 0L;
```

Comment:
Wouldn't that suggest that this user provider never holds any users?

Reply:
I'm assuming that this was copied over from the sakai userprovider as well, not sure why this is this way.

Reply:
The LMS user providers are intended to support looking up users, not "holding" users like the internal user provider does. There's no value in providing an arbitrary user count here.